src/client.js

```diff
@@ -0,0 +1,175 @@
+const cassandra = require('cassandra-driver');
+
+/**
+ * Wrapper for DataStax'x nodeJs cassandra driver.
+ * API reference: https://docs.datastax.com/en/developer/nodejs-driver/4.6/api/
+ */
+
+class Client {
+    /**
+     * Initializes a cassandra client instance.
+     * 
+     * @param {string} clientName: Name to identify a client, could be application/worker/service's name.   
+     * @param {string} keyspace: cassandra's keyspace on which client intends to execute queries.  
+     * @param {string} clientDataCenter: needed to use DefaultLoadBalancingPolicy for load balancing the 
+     *                              requests to cassandra nodes.
+     * @param {Object} clientOptions: Client and query configuration options while setting up the connection. 
+     * @param {Object} emitter: to emit logs. 
+     */
+    constructor(clientName, keyspace, clientDataCenter, clientOptions, emitter) {
```

Comment:
coz I feel it has different requirements. we can have a signature where we can mark default values for `keyspace` and `clientDataCenter`, but I think this is fine too. 
clientDataCenter is needed for using a load balancing provided by datastax's client.
So, I kept them as required args. 
What's wrong in having different signatures though? IMO, they are different entities, they can different signatures.

Reply:
If all of them follow a convention then there is no need to figure out where everything goes. Ex, emitter is always the second parameter, in this however its last

Also we would like to avoid 5 params

Reply:
updated the signature.